src/lib/cms.ts

```diff
@@ -69,3 +68,254 @@ function proxiedRivetClient(productSlug: ProductSlug) {
 
 export default instance
 export { client, proxiedRivetClient }
+
+/**
+ *
+ * NOTE: types defined below are only used here. Refer to ProductSlug and
+ * ProductData from /src/data/types/products.ts if those are needed.
+ *
+ */
```

Comment:
@dstaley we chatted about this briefly on another PR, what do you think about these changes?

Reply:
I think I'd prefer these types to live in `src/types/product.ts` since it'd be a bit weird to import from `lib/cms` when you're trying to access a prop like `algoliaConfig` or `alertBanner`. I know that currently imports from `config/<slug>.json` aren't typed, but being able to do something like the following would be nice:

```ts
import type { ProductData } from 'types/product'
import _config from 'data/nomad.json'

const config = _config as ProductData

// ...access properties like config.algoliaConfig with type safety
```

You're totally right that currently the only file that used this `ProductData` type was `lib/cms.ts`, so the hassle right now might not be worth it, but it'd be nice to have the ability to use it in the future.

Totally fine punting this down the line and worrying about it later though!

Reply:
@dstaley We do actually already do what's in your code snippet in many files in DevDot. 😊

I think where this gets really tricky is that there are properties in the `data/<slug>.json` that .io sites only use, and there are some in DevDot pages that aren't used in .io sites. I don't think we can have one type that they both share because their shapes differ too much. I think we can move the definitions to the existing `src/types/products.ts` as long as we define two different `ProductData` interfaces. I think that probably would best done in a separate PR like you mention so we can have a more focused discussion on it if needed.

Reply:
Yeah, let's explore this a bit more in a future PR then! I think your idea of two separate interfaces is a great idea. That'd even give us some signals when we're using a .io property on a devdot page and vice versa.